app/controllers/declaration/ConfirmationController.scala

```diff
@@ -16,31 +16,101 @@
 
 package controllers.declaration
 
-import controllers.actions.AuthAction
+import scala.concurrent.{ExecutionContext, Future}
+
+import connectors.CustomsDeclareExportsConnector
+import controllers.actions.{AuthAction, VerifiedEmailAction}
+import controllers.declaration.ConfirmationController._
+import controllers.routes.{RejectedNotificationsController, SubmissionsController}
+import handlers.ErrorHandler
+import javax.inject.Inject
+import models.requests.{AuthenticatedRequest, ExportsSessionKeys}
+import play.api.Logging
 import play.api.i18n.I18nSupport
 import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
 import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import views.helpers.Confirmation
 import views.html.declaration.confirmation._
 
-import javax.inject.Inject
-
 class ConfirmationController @Inject()(
   authenticate: AuthAction,
+  verifyEmail: VerifiedEmailAction,
+  customsDeclareExportsConnector: CustomsDeclareExportsConnector,
   mcc: MessagesControllerComponents,
+  errorHandler: ErrorHandler,
   draftConfirmationPage: draft_confirmation_page,
   holdingConfirmationPage: holding_confirmation_page,
   submissionConfirmationPage: submission_confirmation_page
-) extends FrontendController(mcc) with I18nSupport {
+)(implicit ec: ExecutionContext) extends FrontendController(mcc) with I18nSupport with Logging {
 
   val displayDraftConfirmation: Action[AnyContent] = authenticate { implicit request =>
     Ok(draftConfirmationPage())
   }
 
-  val displayHoldingConfirmation: Action[AnyContent] = authenticate { implicit request =>
-    Ok(holdingConfirmationPage())
+  val displayHoldingConfirmation: Action[AnyContent] = authenticate.async { implicit request =>
+    request.getQueryString(js) match {
+
+      // Show page at /holding and wait a few secs.
+      case None =>
+        val holdingUrl = routes.ConfirmationController.displayHoldingConfirmation.url
+        Future.successful(Ok(holdingConfirmationPage(s"$holdingUrl?$js=$Disabled", s"$holdingUrl?$js=$Enabled")))
```

Comment:
Nit: feels a little odd for the controller to pass these two values to the view, couldn't the view just be passes the `holdingUrl` and specialise it for its own needs?